Sources/Apodini/Semantic Model Builder/GraphQL/GraphQLSchemaBuilder.swift

```diff
@@ -0,0 +1,248 @@
+
+
+func graphqlTypeMap(with type: Codable.Type) -> GraphQLScalarType {
+    if (type == String.self) {
+        return GraphQLString
+    } else if (type == Int.self || type == UInt.self) {
+        return GraphQLInt
+    } else if (type == Float.self) {
+        return GraphQLFloat
+    } else if (type == Bool.self) {
+        return GraphQLBoolean
+    }
+    return GraphQLString
+
+}
+
+struct GraphQLRequest: ExporterRequest {
+    var source: Any
+    var args: Map
+    var context: Any
+    var info: GraphQLResolveInfo
+}
+
+
+class GraphQLSchemaBuilder {
+    private var tree = [String: Set<String>]()
+    private var leafContext = [String: AnyConnectionContext<GraphQLInterfaceExporter>]()
+    private var hasIncomingEdge = Set<String>()
+
+
+    // GraphQL Related values
+    private var types = [String: GraphQLObjectType]()
+    private var fields = [String: GraphQLField]()
+    private var args = [String: [String: GraphQLArgument]]()
+    private var responseTypeTree = [String: Node<EnrichedInfo>]()
+
+    private func typeToGraphQL(type: Any.Type) -> GraphQLType? {
+        if (type == String.self) {
+            return GraphQLString
+        }
+        if (type == Int.self || type == UInt.self) {
+            return GraphQLInt
+        }
+        if (type == Bool.self) {
+            return GraphQLBoolean
+        }
+        if (type == Float.self) {
+            return GraphQLFloat
+        }
+        return nil
+    }
+
+    private func responseTypeHandler(for responseTypeHead: Node<EnrichedInfo>) -> GraphQLOutputType {
+        let typeValTemp = self.typeToGraphQL(type: responseTypeHead.value.typeInfo.type)
+
+        if let typeVal = typeValTemp {
+            return typeVal as! GraphQLOutputType
+        }
+
+
+        // Array / Optional
+        if let newResponseHead = try! responseTypeHead.edited(handleArray)?.edited(handleOptional) {
+            if (newResponseHead.value.cardinality == .zeroToMany(.array)) { // Array
+                if let eNode = try? EnrichedInfo.node(newResponseHead.value.typeInfo.type) {
+                    return GraphQLList(responseTypeHandler(for: eNode))
+                }
+            } else if (newResponseHead.value.cardinality == .zeroToOne) { // Optional
+                if let eNode = try? EnrichedInfo.node(newResponseHead.value.typeInfo.type) {
+                    return responseTypeHandler(for: eNode)
+                }
+            }
+        }
+
+        var currentFields = [String: GraphQLField]()
+        for c in responseTypeHead.children {
+            if let propertyInfo = c.value.propertyInfo {
+                currentFields[propertyInfo.name] = GraphQLField(type: responseTypeHandler(for: c), resolve: { source, args, context, info in
+                    return try propertyInfo.get(from: source)
+                })
+            }
+        }
+
+        let typeName = responseTypeHead.value.typeInfo.name
+        return try! GraphQLObjectType(name: typeName, fields: currentFields)
```

Comment:
Don't force try. It's better to throw the error. Then someone can attempt to catch it gracefully. 
For example this statement will crash if someone uses an Empty struct